File: temp class/Event.cpp
#include <iostream>
#include "Event.h"

using namespace std;

Event::Event()
{
    numParticipants = 0;
}

void Event::addEvent()
{

    cout << "Enter Event ID: ";
    cin >> eventID;

    cout << "Enter Event Name: ";
    cin >> eventName;

    cout << "Enter Event Date: ";
    cin >> eventDate;

    cout << "Enter Max Participants: ";
    cin >> maxParticipants;

}

void Event::displayEvent()
{

    cout << "\nEvent ID: " << eventID << endl;
    cout << "Event Name: " << eventName << endl;
    cout << "Event Date: " << eventDate << endl;
    cout << "Participants: " << numParticipants << "/" << maxParticipants << endl;

}

bool Event::isFull()
{

    if(numParticipants >= maxParticipants)
        return true;
    else
        return false;

}

void Event::addParticipant()
{

    numParticipants++;

}

int Event::getEventID()
{

    return eventID;

}

File: classes/event.ts
export class Event {

    private event_id: string;
    private event_name: string;
    private event_date: Date;
    private numofparticipants: number;
    private max_participants: number;


    constructor(

        event_id: string,
        event_name: string,
        event_date: Date,
        max_participants: number,
        numofparticipants: number = 0

    ) {

        this.event_id = event_id;
        this.event_name = event_name;
        this.event_date = event_date;
        this.max_participants = max_participants;
        this.numofparticipants = numofparticipants;

    }



    // Check space

    public hasSpace(): boolean {

        return this.numofparticipants < this.max_participants;

    }



    // Add participant

    public addParticipant(): boolean {

        if (this.hasSpace()) {

            this.numofparticipants++;

            return true;

        }

        return false;

    }



    // Display event info

    public displayInfo(): string {

        return `
Event ID: ${this.event_id}
Event Name: ${this.event_name}
Date: ${this.event_date}
Participants: ${this.numofparticipants}
Max Participants: ${this.max_participants}
`;

    }

}